File: Examples/AdFoxMediationInterstitialExample/AdFoxMediationInterstitialExample/ViewController.m
/*
 *  ViewController.m
 *
 * This file is a part of the Yandex Advertising Network.
 *
 * Version for iOS © 2017 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://legal.yandex.com/partner_ch/
 */

#import "ViewController.h"
#import <YandexMobileAds/YandexMobileAds.h>

@interface ViewController () <YMAInterstitialDelegate>

@property (nonatomic, strong) YMAInterstitialController *interstitialController;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    // Replace demo R-M-206876-12 with actual Block ID.
    self.interstitialController = [[YMAInterstitialController alloc] initWithBlockID:@"R-M-206876-12"];
    self.interstitialController.delegate = self;
}

- (IBAction)loadInterstitial
{
    // Replace demo parameters with actual parameters.
    NSMutableDictionary *parameters = [[NSMutableDictionary alloc] init];
    parameters[@"adf_ownerid"] = @"168627";
    parameters[@"adf_p1"] = @"bxvfr";
    parameters[@"adf_p2"] = @"fhmf";

    YMAAdRequest *adRequest = [[YMAAdRequest alloc] initWithLocation:nil
                                                        contextQuery:nil
                                                         contextTags:nil
                                                          parameters:parameters];
    [self.interstitialController loadWithRequest:adRequest];
}

- (IBAction)presentInterstitial
{
    [self.interstitialController presentInterstitialFromViewController:self];
}

#pragma mark - YMAInterstitialDelegate

- (void)interstitialDidLoadAd:(YMAInterstitialController *)interstitial
{
    NSLog(@"Loaded");
}

- (void)interstitialDidFailToLoadAd:(YMAInterstitialController *)interstitial error:(NSError *)error
{
    NSLog(@"Loading failed. Error: %@", error);
}

- (void)interstitialWillLeaveApplication:(YMAInterstitialController *)interstitial
{
    NSLog(@"Will leave application");
}

- (void)interstitialDidFailToPresentAd:(YMAInterstitialController *)interstitial error:(NSError *)error
{
    NSLog(@"Failed to present interstitial. Error: %@", error);
}

- (void)interstitialWillAppear:(YMAInterstitialController *)interstitial
{
    NSLog(@"Interstitial will appear");
}

- (void)interstitialDidAppear:(YMAInterstitialController *)interstitial
{
    NSLog(@"Interstitial did appear");
}

- (void)interstitialWillDisappear:(YMAInterstitialController *)interstitial
{
    NSLog(@"Interstitial will disappear");
}

- (void)interstitialDidDisappear:(YMAInterstitialController *)interstitial
{
    NSLog(@"Interstitial did disappear");
}

- (void)interstitialWillPresentScreen:(UIViewController *)webBrowser
{
    NSLog(@"Interstitial will present screen");
}

@end

File: Examples/Swift/AdFoxMediationInterstitialExample/AdFoxMediationInterstitialExample/ViewController.swift
/*
 * Version for iOS © 2015–2019 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

import UIKit
import YandexMobileAds

class ViewController: UIViewController {
    
    @IBOutlet weak var showButton: UIButton!

    private let adMobBlockID = "adf-279013/975869";
    private let facebookBlockID = "adf-279013/975872";
    private let moPubBlockID = "adf-279013/975870";
    private let myTargetBlockID = "adf-279013/975871";
    private let startAppBlockID = "adf-279013/1006406"
    private let unityAdsBlockID = "adf-279013/1006439"
    private let yandexBlockID = "adf-279013/975873";

    var interstitialAd: YMAInterstitialController!

    @IBAction func loadAd(_ sender: UIButton) {
        /*
         Replace demo AdMobBlockID with actual Block ID.
         Following demo block ids may be used for testing:
         AdMob mediation: adMobBlockID
         Facebook mediation: facebookBlockID
         MoPub mediation: moPubBlockID
         MyTarget mediation: myTargetBlockID
         StartApp mediation: startAppBlockID
         UnityAds mediation: unityAdsBlockID
         Yandex: yandexBlockID
         */
        interstitialAd = YMAInterstitialController(blockID: adMobBlockID)
        interstitialAd.delegate = self
        interstitialAd.load()
    }

    @IBAction func presentAd(_ sender: UIButton) {
        interstitialAd.presentInterstitial(from: self)
    }
}

extension ViewController: YMAInterstitialDelegate {
    func interstitialDidLoadAd(_ interstitial: YMAInterstitialController!) {
        print("Interstitial did load ad")
        showButton.isUserInteractionEnabled = true
    }

    func interstitialDidFail(toPresentAd interstitial: YMAInterstitialController!, error: Error!) {
        print("Interstitial did fail to present ad with error: \(String(describing: error))")
    }

    func interstitialDidFail(toLoadAd interstitial: YMAInterstitialController!, error: Error!) {
        print("Interstitial did fail to load ad with error: \(String(describing: error))")
    }
}

